src/toil_lib/tools/aligners.py

```diff
@@ -103,47 +104,58 @@ def run_bwakit(job, config, sort=True, trim=False):
     :rtype: str
     """
     work_dir = job.fileStore.getLocalTempDir()
-    file_names = ['r1.fq.gz', 'ref.fa.fai', 'ref.fa', 'ref.fa.amb', 'ref.fa.ann',
-                  'ref.fa.bwt', 'ref.fa.pac', 'ref.fa.sa']
-    ids = [config.r1, config.ref, config.fai, config.amb, config.ann, config.bwt, config.pac, config.sa]
+    inputs = {'ref.fa': config.ref,
+              'ref.fa.fai': config.fai,
+              'ref.fa.amb': config.amb,
+              'ref.fa.ann': config.ann,
+              'ref.fa.bwt': config.bwt,
+              'ref.fa.pac': config.pac,
+              'ref.fa.sa': config.sa}
+    samples = []
     # If a fastq pair was provided
+    if getattr(config, 'r1', None):
+        inputs['input.1.fq.gz'] = config.r1
+        samples.append('input.1.fq.gz')
     if getattr(config, 'r2', None):
-        file_names.insert(1, 'r2.fq.gz')
-        ids.insert(1, config.r2)
+        inputs['input.2.fq.gz'] = config.r2
+        samples.append('input.2.fq.gz')
+    if getattr(config, 'bam', None):
+        inputs['input.bam'] = config.bam
+        samples.append('input.bam')
     # If an alt file was provided
     if getattr(config, 'alt', None):
-        file_names.append('ref.fa.alt')
-        ids.append(config.alt)
-    for fileStoreID, name in zip(ids, file_names):
+        inputs['ref.fa.alt'] = config.alt
+    for name, fileStoreID in inputs.iteritems():
         job.fileStore.readGlobalFile(fileStoreID, os.path.join(work_dir, name))
     # If a read group line was provided
     if getattr(config, 'rg_line', None):
         rg = config.rg_line
     # Otherwise, generate a read group line to place in the BAM.
-    else:
+    elif all(getattr(config, elem, None) for elem in ['library', 'platform', 'program_unit', 'uuid']):
         rg = "@RG\\tID:{0}".format(config.uuid)  # '\' character is escaped so bwakit gets passed '\t' properly
         rg_attributes = [config.library, config.platform, config.program_unit, config.uuid]
         for tag, info in zip(['LB', 'PL', 'PU', 'SM'], rg_attributes):
             rg += '\\t{0}:{1}'.format(tag, info)
+    else:
+        rg = None
```

Comment:
IMO, there's no case where we should allow an unset `@RG` line, because it breaks tons of downstream stuff. Is this change meant to allow `bwakit` to get the `@RG` line(s) from a BAM file header? If so, this is OK, but this fork of the code should require that a BAM file is provided.